middleware/oauth.global.ts

```diff
@@ -24,7 +24,78 @@ export default defineNuxtRouteMiddleware(async (to) => {
       }
     }
   }
+  console.log("to.path", to.path);
+  // Check if this is a dataset route that might have public access
+  // Check if this is a dataset route that might have public access
+  const isDatasetRoute =
+    to.path.startsWith("/alerts/") ||
+    to.path.startsWith("/gallery/") ||
+    to.path.startsWith("/map/");
 
+  if (isDatasetRoute) {
+    console.log("isDatasetRoute", isDatasetRoute, "for path:", to.path);
+    try {
+      const {
+        public: { appApiKey },
+      } = useRuntimeConfig();
+      const headers = { "x-api-key": appApiKey };
+
+      const data = (await $fetch("/api/config", { headers })) as unknown as [
+        { [key: string]: { routeLevelPermission?: RouteLevelPermission } },
+        string[],
+      ];
+
+      // Extract table name from path (e.g., "/alerts/tableName" -> "tableName")
+      const pathParts = to.path.split("/");
+      const tableName = pathParts[pathParts.length - 1];
+
+      const viewConfig = data?.[0]?.[tableName];
+      const permission: RouteLevelPermission =
+        viewConfig?.routeLevelPermission ?? "anyone"; // Default to anyone if not set
```

Comment:
```suggestion
    const [tableConfig] = (await $fetch("/api/config", { headers })) as [
      Record<string, { routeLevelPermission?: RouteLevelPermission }>,
      string[],
    ];

    // Extract the table name from the last part of the path
    const tableName = to.path.split("/").pop()!;
    const permission: RouteLevelPermission =
      tableConfig?.[tableName]?.routeLevelPermission ?? "member";
```

Just to make this a little easier to follow.